app/views/decidim/anonymous_codes/admin/code_groups/index.html.erb

```diff
@@ -0,0 +1,48 @@
+<div class="card">
+  <div class="card-divider">
+    <h2 class="card-title">
+      <%= t("code_groups.index.title", scope: "decidim.anonymous_codes.admin") %>
+      <a class="button tiny button--title new-access" href="<%= new_code_group_path %>"><%= t("code_groups.index.new_access_code_group_button", scope: "decidim.anonymous_codes.admin") %></a>
+    </h2>
+  </div>
+  <div class="card-section">
+    <% if @groups.any? %>
+      <div class="table-scroll">
+        <table class="table-list">
+          <thead>
+            <tr>
+              <th><%= t("code_groups.index.header_table_title", scope: "decidim.anonymous_codes.admin") %></th>
+              <th><%= t("code_groups.index.expires_at", scope: "decidim.anonymous_codes.admin") %></th>
+              <th><%= t("code_groups.index.active", scope: "decidim.anonymous_codes.admin") %></th>
+              <th><%= t("code_groups.index.num_of_tockens", scope: "decidim.anonymous_codes.admin") %></th>
+              <th><%= t("code_groups.index.max_reuses", scope: "decidim.anonymous_codes.admin") %></th>
+              <th></th>
+            </tr>
+          </thead>
+          <tbody>
+            <% groups.each do |group| %>
+              <tr>
+                <td><%= translated_attribute(group.title) %></td>
+                <td><%= l(group.expires_at, format: :decidim_short) %></td>
```

Comment:
```suggestion
                <td><%= group.expires_at.present? ? "<span class=\"text-#{ group.expired? ? 'warning' : ''}\">#{l(group.expires_at, format: :decidim_short)}</span>" : t('.never') %></td>
```
be aware that if you don't define a date, this will fail (we should have a system test case for it). With this condition should be solved. Of course, add the proper I18n text.

Also note that we can decorate here the case when is expired, for that we can add to the `group.rb` model the following method:

app/models/decidim/anonymous_codes/group.rb
```ruby
      def expired?
        expires_at.present? && expires_at < Time.current
      end
```
Also, remove it from token.rb and add it to the `delegate` helper:

app/models/decidim/anonymous_codes/token.rb
```ruby
    delegate :active?, :expired? to: :group
```

Reply:
Ok, done.